### ISSUE TITLE
Review of restDecay0

### ISSUE THREAD

**Issue by jgalan** [rest-for-physics/framework#6]
It seems to me that the class found at `packages/restDecay0` would be better part of the geant4 library.

This would require a renaming: `TRestParticleCollectionDecay0` --> `TRestGeant4ParticleCollectionDecay0`.

Then, the `geant4` library should have just a compilation option to enable this. Just as we enable other features as `REST_SQL` or `REST_EVE`. We could have another `REST_DECAY0` that enables this feature and makes `RestGeant4` to link against `Decay0`. This kind of options I would refer it as a `plugin`, it is a class that extends the framework or library functionality.

A packages is more a standalone program, with its own executable, that links and makes use of REST.

**Comment by nkx111** [comment]
I agree. In 2.3 point of view, these classes shall better placed together with `TRestGeant4ParticleCollection`, `TRestGeant4PrimaryGenerator`, etc., inside `RestGeant4` library. Both of them are standalone (installed separately) with REST Framework

**Comment by jgalan** [comment]
Ok, I have started the migration. I moved the objects to the geant4 library and re-adapted the CMakeLists accordingly. However, I have difficulties to compile.

I tried with 1.0.0 and 1.0.9/development that I found at:
https://github.com/BxCppDev/bxdecay0

With 1.0.9 `find_packages` was not working properly. 

But the real problem is when I try to compile, I get the following error:

```
/Users/javi/rest-framework/source/libraries/geant4/src/TRestGeant4ParticleCollectionDecay0.cxx:118:55: error: no member named
      'MODEBB_4' in namespace 'bxdecay0'
            fDecay0Model.set_decay_dbd_mode(bxdecay0::MODEBB_4);
                                            ~~~~~~~~~~^
```
Same error with 1.0.0 and 1.0.9

**Comment by nkx111** [comment]
The compilation problem is fixed at aa51394!
